File: app/src/main/java/com/example/pearvideoclient/screen/ScreenFragment.java
package com.example.pearvideoclient.screen;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.pearvideoclient.MyApplication;
import com.example.pearvideoclient.R;
import com.example.pearvideoclient.content.VideoPlayerIJK;
import com.example.pearvideoclient.entity.PaikeFineVideoBean;
import com.example.pearvideoclient.utils.MyToast;

import java.util.ArrayList;
import java.util.List;

/**
 * @author zhujiaqqq
 * @date 2019-07-11
 */
public class ScreenFragment extends Fragment implements ScreenContract.View {
    private VideoPlayerIJK mVideoPlayer;
    private RecyclerView mRvVideoList;
    private ImageView mIvVideoImg;

    private Context mContext;
    private VideoAdapter mVideoAdapter;

    private ScreenContract.Presenter mPresenter;

    private View.OnClickListener mClickListener = v -> {
        switch (v.getId()) {
            case R.id.tv_start_video:
                break;
            case R.id.tv_see_detail:
                break;
            case R.id.tv_remark_list:
                break;
            case R.id.tv_my_video:
                break;
            default:
                break;
        }
    };
    private PaikeFineVideoBean.ConfigInfoBean mConfigInfo;

    public static ScreenFragment newInstance() {
        Bundle bundle = new Bundle();
        ScreenFragment fragment = new ScreenFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_screen, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initView(view);
        initData();
    }

    private void initView(View view) {
        mVideoPlayer = view.findViewById(R.id.video_player_tech);
        mRvVideoList = view.findViewById(R.id.rv_video_list);
        mIvVideoImg = view.findViewById(R.id.iv_video_img);

        view.findViewById(R.id.tv_start_video).setOnClickListener(mClickListener);
        view.findViewById(R.id.tv_see_detail).setOnClickListener(mClickListener);
        view.findViewById(R.id.tv_remark_list).setOnClickListener(mClickListener);
        view.findViewById(R.id.tv_my_video).setOnClickListener(mClickListener);
    }

    private void initData() {
        new ScreenPresenter(this);
        mContext = getActivity();
        mRvVideoList.setLayoutManager(new GridLayoutManager(mContext, 2));
        mVideoAdapter = new VideoAdapter(R.layout.adapter_paike_video_item, new ArrayList<>());
        mVideoAdapter.setOnItemChildClickListener((adapter, view, position) -> {
            if (view.getId() == R.id.rl_parent) {
                // TODO: 2019-08-26

            }
        });
        mRvVideoList.setAdapter(mVideoAdapter);
        mPresenter.subscribe();
    }

    @Override
    public void onDestroy() {
        mPresenter.unsubscribe();
        super.onDestroy();
    }

    @Override
    public void showPaikeVideos(List<PaikeFineVideoBean.VideoBean> data) {
        mVideoAdapter.replaceData(data);
    }

    @Override
    public void showPaikeInfo(PaikeFineVideoBean.ConfigInfoBean data) {
        mConfigInfo = data;
        mIvVideoImg.setImageResource(R.drawable.ic_paike_img);
    }

    @Override
    public void setPresenter(ScreenContract.Presenter presenter) {
        mPresenter = presenter;
    }

    @Override
    public void showLoading() {
        //
    }

    @Override
    public void cancelLoading() {
        //
    }

    @Override
    public void showErrorToast(String loadingFail) {
        MyToast.getInstance(MyApplication.getInstance()).show(loadingFail, 3000);
    }
}

File: app/src/main/java/com/example/pearvideoclient/screen/ScreenFragment.kt
package com.example.pearvideoclient.screen

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import com.example.pearvideoclient.MyApplication
import com.example.pearvideoclient.R
import com.example.pearvideoclient.entity.PaikeFineVideoBean
import com.example.pearvideoclient.utils.MyToast
import com.example.route.Route
import kotlinx.android.synthetic.main.fragment_screen.*
import java.util.*

/**
 * @author zhujiaqqq
 * @date 2019-07-11
 */
class ScreenFragment : Fragment(), ScreenContract.View {

    private var mContext: Context? = null
    private var mVideoAdapter: VideoAdapter? = null

    private var mPresenter: ScreenContract.Presenter? = null

    private val mClickListener = View.OnClickListener { v ->
        when (v.id) {
            R.id.tv_start_video -> Route.getInstance().jumpActivity("videolib/recordVideo", null)
            R.id.tv_see_detail -> {
            }
            R.id.tv_remark_list -> {
            }
            R.id.tv_my_video -> {
            }
            else -> {
            }
        }
    }
    private var mConfigInfo: PaikeFineVideoBean.ConfigInfoBean? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_screen, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
        initData()
    }

    private fun initView() {
        tv_start_video.setOnClickListener(mClickListener)
        tv_see_detail.setOnClickListener(mClickListener)
        tv_remark_list.setOnClickListener(mClickListener)
        tv_my_video.setOnClickListener(mClickListener)
    }

    private fun initData() {
        ScreenPresenter(this)
        mContext = activity
        rv_video_list.layoutManager = GridLayoutManager(mContext, 2)
        mVideoAdapter = VideoAdapter(R.layout.adapter_paike_video_item, ArrayList<PaikeFineVideoBean.VideoBean>())
        mVideoAdapter!!.setOnItemChildClickListener { _, view, _ ->
            if (view.id == R.id.rl_parent) {
                // TODO: 2019-08-26

            }
        }
        rv_video_list.adapter = mVideoAdapter
        mPresenter!!.subscribe()
    }

    override fun onDestroy() {
        mPresenter!!.unsubscribe()
        super.onDestroy()
        MyApplication.getRefWatcher(activity!!)!!.watch(this)
    }

    override fun showPaikeVideos(data: List<PaikeFineVideoBean.VideoBean>) {
        mVideoAdapter!!.replaceData(data)
    }

    override fun showPaikeInfo(data: PaikeFineVideoBean.ConfigInfoBean) {
        mConfigInfo = data
        iv_video_img.setImageResource(R.drawable.ic_paike_img)
    }

    override fun setPresenter(presenter: ScreenContract.Presenter) {
        mPresenter = presenter
    }

    override fun showLoading() {
        //
    }

    override fun cancelLoading() {
        //
    }

    override fun showErrorToast(loadingFail: String) {
        MyToast.getInstance(MyApplication.instance).show(loadingFail, 3000)
    }

    companion object {

        fun newInstance(): ScreenFragment {
            val bundle = Bundle()
            val fragment = ScreenFragment()
            fragment.arguments = bundle
            return fragment
        }
    }
}
